security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -460,7 +463,8 @@ func (sc *SecretManagerClient) readFileWithTimeout(path string) ([]byte, error)
 	}
 }
 
-func (sc *SecretManagerClient) generateFileSecret(resourceName string) (bool, *security.SecretItem, error) {
+func (sc *SecretManagerClient) generateFileSecret(resourceName string, caRootPath string) (bool, *security.SecretItem, error) {
```

Comment:
Should we pass in caRootPath or set it in the previous function and use the getter here?